src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -101,14 +121,17 @@ export class ElasticsearchService implements CoreService<ElasticsearchServiceSet
 
     this.subscription = clients$.connect();
 
+    const config = await this.config$.pipe(first()).toPromise();
+
     return {
       legacy: { config$: clients$.pipe(map(clients => clients.config)) },
 
       adminClient$: clients$.pipe(map(clients => clients.adminClient)),
       dataClient$: clients$.pipe(map(clients => clients.dataClient)),
 
-      createClient: (type: string, clientConfig: ElasticsearchClientConfig) => {
-        return this.createClusterClient(type, clientConfig, deps.http.auth.getAuthHeaders);
+      createClient: (type: string, clientConfig: Partial<ElasticsearchClientConfig> = {}) => {
+        const finalConfig = merge({}, config, clientConfig);
```

Comment:
it's not a full equivalent of 
```js
{
  ...esConfig,
  ...clientConfig
}
```
but I'd expect that user may want to re-write only a part of complex `ssl` config instead of declaring the full version of it.